### PR TITLE
Raise ASN1::ASN1Error

### PR DESCRIPTION
**Pull request by thekuwayama**
Hi!

This PR is in order to raise the appropriate exception.
if certificate extension ASN1 format is not appropriate, not `NoMethodError` but `ASN1Error` is raised.

### CODE REVIEW

**Comment by thekuwayama** [comment]
Probably, it is forgot `new`.

```
$ irb
irb(main):001:0> require 'openssl'
=> true
irb(main):002:0> raise OpenSSL::ASN1::ASN1Error 'invalid extension'
Traceback (most recent call last):
        4: from /Users/thekuwayama/.rbenv/versions/2.6.5/bin/irb:23:in `<main>'
        3: from /Users/thekuwayama/.rbenv/versions/2.6.5/bin/irb:23:in `load'
        2: from /Users/thekuwayama/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        1: from (irb):2
NoMethodError (undefined method `ASN1Error' for OpenSSL::ASN1:Module)
```

File: lib/openssl/x509.rb [hunk 1]
```diff
@@ -99,7 +99,7 @@ def subject_key_identifier
 
           ski_asn1 = ASN1.decode(ext.value_der)
           if ext.critical? || ski_asn1.tag_class != :UNIVERSAL || ski_asn1.tag != ASN1::OCTET_STRING
-            raise ASN1::ASN1Error "invalid extension"
+            raise ASN1::ASN1Error.new("invalid extension")
```
Comment (ioquatix) [inline lib/openssl/x509.rb]: Can you change this to `raise ASN1::ASN1Error, "message"` - it's a little bit more idiomatic than `.new("message")`.

**Comment by ioquatix** [comment]
Can you also add an entry to the changelog following the existing format.

**Comment by thekuwayama** [comment]
@ioquatix This PR just fix `Certificate#{authority_key_identifier, subject_key_identifier, crl_uris}` that is not yet released so this does not add functionality from version 2.1.2.
I think no need to add an entry to the changelog. Check it.

**Comment by ioquatix** [comment]
Sorry, yeah, I should have realised that. Looks good.

**Comment by thekuwayama** [comment]
Thank you for reviewing and checking!!

**Pull request merged**